packages/client/src/helpers.ts

```diff
@@ -7,13 +7,11 @@ export function buildQueryString(parameters: Record<string, any>): string {
         return '';
     }
 
-    let query = '?';
+    let query = '';
     for (let i = 0; i < keys.length; i++) {
         const key = keys[i];
-        query += key + '=' + encodeURIComponent(parameters[key]);
-
-        if (i < keys.length - 1) {
-            query += '&';
+        if (parameters[key] !== undefined) {
+            query += `${query.length === 0 ? '?' : '&'}${key}=${encodeURIComponent(parameters[key])}`;
```

Comment:
Having a ternary with string literals inside of a template string feels very hard to read. Perhaps instead of concatenating onto query repeatedly, it would be simpler if we rewrite it like
```javascript
const queryParams = Object.entries(parameters).
    filter(([key, value]) => value !== undefined).
    map(([key, value]) => `${key}=${encodeURIComponent(parameters[key])}`);

return queryParams.length > 0 ? '&' + queryParams.join('&') : '';
```

Reply:
i will think of something .. although your approach seems fine already.